app/controllers/pair_requests/acceptances_controller.rb

```diff
@@ -3,9 +3,16 @@ def create
     @pair_request = PairRequest.find(params[:pair_request_id])
     authorize @pair_request, policy_class: PairRequest::AcceptancePolicy
     @pair_request.accepted!
-    PairRequest::AcceptanceNotification.with(pair_request: @pair_request).deliver(@pair_request.author)
+    send_notifications
 
     flash[:notice] = 'You have accepted this pair request.'
     redirect_to pair_request_path(@pair_request)
   end
+
+  private
+
+  def send_notifications
+    PairRequest::AcceptanceNotification.with(pair_request: @pair_request).deliver(@pair_request.author)
+    PairRequest::CompletionReminderNotification.with(pair_request: @pair_request).deliver(@pair_request.author)
```

Comment:
Just one question related to this. It seems like there is a delay set. Is the `if: :pair_request_accepted?` only triggered once it trys to send the email?

So the flow is:
1. invitee accepts pair request
2. creates the send notifcations for the author
3. the completion one is only triggered once it is after the delay set. If at this point the pair request is still in an acceptance, it will send a reminder, otherwise it will just noop?

Reply:
Yes exactly. The condition is used when it comes time to send the email. If the condition is false when it's time to send it, nothing happens. So users who have already marked the request as completed (or they've canceled or whatever) won't get the email.

Reply:
I was initially a bit confused about this too but parts  of the Noticed doc make it clear what's happening. Like [here in the README](https://github.com/excid3/noticed/tree/master#fallback-notifications) he shows a pattern for fallback notifications that also depends on the conditions being evaluated on a delay.